File: WoofAndroid/data/datasource/src/testFixtures/kotlin/com/perrystreet/woof/datasource/woofs/WoofsDataSourceFactory.kt
package com.perrystreet.woof.datasource.woofs

import org.koin.test.KoinTest
import org.koin.test.inject

class WoofsDataSourceFactory : KoinTest {
    private val dataSource: FakeWoofsDataSource by inject()

    fun withSendWoofError() = apply {
        dataSource.sendWoofError = IllegalStateException("Could not send woof")
    }
}

File: WoofIOS/SwiftPackages/Data/DataSource/Sources/DataSourceFakes/Woofs/WoofsDataSourceFactory.swift
import DataSource
import Swinject
import SwinjectAutoregistration

public final class WoofsDataSourceFactory {
    private let dataSource: FakeWoofsDataSource

    public init(_ container: Container) {
        dataSource = container~>
    }

    @discardableResult
    public func withSendWoofError() -> Self {
        dataSource.sendWoofError = .unavailable("Could not send woof")
        return self
    }
}
